### DISCUSSION TITLE
System.Net.Http.HttpConnectionPool.CleanCacheAndDisposeIfUnused() disposing http connection

### DISCUSSION THREAD

**Discussion by eyagovbusiness** [DSharpPlus/DSharpPlus#1478] [Support]
I am using your library in a ASP.NET and after running few minutes the Api, every single exceptions are thrown about trying to access disposed object. I was digging a bit into the Strack trace and the parallel stacks and I found in System.Net.Http.HttpConnectionPool.CleanCacheAndDisposeIfUnused() the HttpConnection(HttpConnectionPool https://discord.com:443) is being disposed. May you take a look into this?
![image](https://user-images.githubusercontent.com/124583064/221389564-2ca21f2a-e252-47bb-aa9a-307a8fde8267.png)
![image](https://user-images.githubusercontent.com/124583064/221389590-9296c21d-da32-419e-91df-766a28ee34cf.png)

Here the object is disposed.

![image](https://user-images.githubusercontent.com/124583064/221389671-783bcb89-52c6-41ea-b75a-298b68b7135e.png)
Here the exception is thrown.

![image](https://user-images.githubusercontent.com/124583064/221389772-b2f6375d-c936-403c-b36b-3c56b24a5757.png)

After this my program increases the RAM usage by 40 MB indefinitely, which matches the read buffer size of the disposed HttpConnection, so it looks also like the Garbage Collector is not cleaning this.

 I can provide more info if needed :)

**Comment by VelvetToroyashi** [comment]
I'm more or less curious how things are wired up (is this project OSS?), because the library isn't *really* designed to be used in conjunction with ASP.NET (we implore you to use the generic host for runtime coddling)

**Reply by eyagovbusiness** [reply]
Also this, but if support for ASP.NET was not in mind for the project I can understand this may happen. In case you want to take a look the project is available on my GitHub, you will need to pull my custom libraries in order to make it work. Currently I'm running my api in a docker container with alpine but you can use one of my configured launchsettings and start it without docker. 
![image](https://user-images.githubusercontent.com/124583064/221390918-d96e8599-ca05-47ed-bb04-477ab82f1d3c.png)

It's just a personal project that I started not long ago and I am making it for my portfolio and a group of friends. Its public so feel free to download it :)

**Reply by VelvetToroyashi** [reply]
Odd. From what I can tell this should be a non-issue. Perhaps ask on the [C# Discord](https://discord.gg/csharp)?

**Reply by eyagovbusiness** [reply]
Ok, I will try to continue researching and I will also join that discord server, thank you for your time. :)
